git-branchless-hook/src/lib.rs

```diff
@@ -42,6 +42,41 @@ pub use lib::core::rewrite::rewrite_hooks::{
     hook_skip_upstream_applied_commit,
 };
 
+/// A component that will find the hash for a given reference transaction OID.
+/// Reference transaction OIDs may be one of:
+///
+/// * An up-to-40-hex-digit hash string
+/// * A symbolic ref, in which case, it will start with "ref:"
+///
+/// See: https://github.com/git/git/commit/a8ae923f85da6434c3faf9c39719d6d5e5c77e65
+#[derive(Debug)]
+struct TransactionReferenceFinder<'a> {
```

Comment:
suggestion: I don't think we'll need a `struct`; it should be fine to just have a free function that accepts a `&Repo` and reference name (or possibly elide it altogether if it becomes unnecessary as part of other restructuring).

suggestion: Put it in the `reference_transaction` module somewhere near `parse_reference_transaction_line` (unless it's already there and I just can't read the Github code review rendering).